api/v1/checks.go

```diff
@@ -755,9 +892,58 @@ type AWSConnection struct {
 	UsePathStyle bool `yaml:"usePathStyle,omitempty" json:"usePathStyle,omitempty"`
 }
 
+// populateFromConnection attempts to find the connection by name
+// and populate the endpoint, accessKey and secretKey.
+func (t *AWSConnection) populateFromConnection(ctx context.Context, db *gorm.DB, k8sClient kubernetes.Interface, namespace string) error {
+	connection, err := duty.HydratedConnectionByURL(ctx, db, k8sClient, namespace, t.ConnectionName)
+	if err != nil {
+		return err
+	}
+
+	if connection == nil {
```

Comment:
Isn't this an expected condition = if connectionName != "" and connection == nil && err == nil. then is connection not found, but if it is connectionName == "" then its a noop

Reply:
There was a typo in `.Populate()` . I think that should address this.